theano/tensor/nnet/tests/test_nnet.py

```diff
@@ -281,7 +283,7 @@ def myfunc(x):
         # We set step to 0.1 because for big values we need a big epsilon
         utt.verify_grad(myfunc, [a], eps=0.1, mode=m)
         f = theano.function([], myfunc(a))
-        self.assertTrue(hasattr(f.maker.fgraph.outputs[0].tag, 'trace'))
+        assert check_stack_trace(f, ops_to_check='last')
```

Comment:
The problem here is that `a` is a constant, so `myfunc(a)` gets constant-folded to `DeepCopyOp(TensorConstant(...))`. And since the constant folding seems to happen before the logsoftmax gets inserted (which is another issue entirely), that constant is full of `-inf`...

I think the solution here is to compile a function from a shared variable (not a constant instead). Something like this seems to work:
```python
sa = theano.shared(a)
f = theano.function([], myfunc(sa))
self.assertTrue(check_stack_trace(f, ops_to_check='all'))
```